test/JavaEchoServerTest.java

```diff
@@ -10,22 +11,40 @@
 public class JavaEchoServerTest {
 
     private ByteArrayOutputStream stdOut;
+    private ByteArrayInputStream stdIn;
 
     @Before
     public void setUp() throws Exception {
         stdOut = new ByteArrayOutputStream();
         System.setOut(new PrintStream(stdOut));
+
     }
 
     @After
     public void tearDown() throws Exception {
         System.setOut(null);
```

Comment:
I love that you've found a way to get a fake being used. Can you find a way to avoid having to set global state for the sake of your tests?